core/src/com/unciv/logic/city/CityConstructions.kt

```diff
@@ -503,9 +503,25 @@ class CityConstructions {
         if (!cityInfo.civInfo.gameInfo.gameParameters.godMode) {
             val construction = getConstruction(constructionName)
             if (construction is PerpetualConstruction) return false
-            val constructionCost = (construction as INonPerpetualConstruction).getStatBuyCost(cityInfo, stat)
+            val constructionCost =
+                (construction as INonPerpetualConstruction).getStatBuyCost(cityInfo, stat)
```

Comment:
Wait - if PerpetualConstruction and INonPerpetualConstruction are mutually exclusive, inverting that test in 505 should make the cast redundant... ?